lstchain/reco/utils.py

```diff
@@ -452,12 +452,13 @@ def filter_events(events,
     for k in filters.keys():
         if k in events.columns:
             filter = filter & (events[k] >= filters[k][0]) & (events[k] <= filters[k][1])
+    if finite_params is not None:
+        for k in finite_params:
+            if k in events.columns:
+                filter = filter & np.isfinite(events[k])
```

Comment:
You can do this without the loop. Also it might be helpful to log which columns contain nans and how many.

See:
https://github.com/fact-project/aict-tools/blob/5619074aef9a26063ca41b3c028664a0b3b8a912/aict_tools/preprocessing.py#L24-L35

Reply:
If you also want to filter out `infs` (we deal differently with those, see https://github.com/fact-project/aict-tools/blob/5619074aef9a26063ca41b3c028664a0b3b8a912/aict_tools/preprocessing.py#L14), do:

```
with pd.option_context('mode.use_inf_as_null', True):
    nans = df.isnull()
    filter &= ~nans.any(axis=1)
```

Reply:
done, thanks for the suggestions